ios/Classes/Models/HMSPIPAction.swift

```diff
@@ -0,0 +1,229 @@
+        
+        case "change_text_pip":
+            changeText(call, result, hmsSDK)
+            
+        case "destroy_pip":
+            disposePIP(result)
+
+        default:
+            result(FlutterMethodNotImplemented)
+        }
+    }
+    
+    static func setupPIP(_ call: FlutterMethodCall, _ result: @escaping FlutterResult, _ hmsSDK: HMSSDK?,_ swiftHmssdkFlutterPlugin: SwiftHmssdkFlutterPlugin) {
+        
+        guard AVPictureInPictureController.isPictureInPictureSupported() else {
+            result(HMSErrorExtension.getError("\(#function) PIP is not supported"))
+            return }
+        
+        guard let uiView = UIApplication.shared.keyWindow?.rootViewController?.view else {
+            result(HMSErrorExtension.getError("\(#function) Failed to setup PIP"))
+            return }
+            
+        let pipVideoCallViewController = AVPictureInPictureVideoCallViewController()
+        
+        self.pipVideoCallViewController = pipVideoCallViewController
+           
+        model = PiPModel()
+        model!.track = hmsSDK?.localPeer?.videoTrack
+        model!.pipViewEnabled = true
+        model!.text = hmsSDK?.localPeer?.name
+        
+        let arguments = call.arguments as! [AnyHashable: Any]
+        
+        if let scaleType = arguments["scale_type"] as? Int {
+            model?.scaleType = getViewContentMode(scaleType)
+        }
+        
+        if let color = arguments["color"] as? [Int] {
+            print(#function,color)
+            let colour = Color(red: CGFloat(color[0])/255, green: CGFloat(color[1])/255, blue: CGFloat(color[2])/255)
+            model!.color = colour
+        }
+        
+        let controller = UIHostingController(rootView: PiPView(model: model!))
+        
+        pipVideoCallViewController.view.addConstrained(subview: controller.view)
+                
+    
+        if let ratio = arguments["ratio"] as? [Int], ratio.count == 2 {
+            pipVideoCallViewController.preferredContentSize = CGSize(width: ratio[1], height: ratio[0])
+        } else {
+            pipVideoCallViewController.preferredContentSize = CGSize(width: uiView.frame.size.width, height: uiView.frame.size.height) 
+        }
+        
+        let pipContentSource = AVPictureInPictureController.ContentSource(
+            activeVideoCallSourceView: uiView,
+            contentViewController: pipVideoCallViewController)
+       
+        pipController = AVPictureInPictureController(contentSource: pipContentSource)
+        
+        pipController?.delegate = swiftHmssdkFlutterPlugin
+        
+        if let autoEnterPIP = arguments["auto_enter_pip"] as? Bool {
+            pipController?.canStartPictureInPictureAutomaticallyFromInline = autoEnterPIP
+        }
+
+        NotificationCenter.default.addObserver(forName: UIApplication.didBecomeActiveNotification,
+                                               object: nil, queue: .main) { [self] _ in
```

Comment:
should capture self as `weak`
```suggestion
                                               object: nil, queue: .main) { [weak self] _ in
```